lib/components/molecules/Typeahead/Typeahead.js

```diff
@@ -0,0 +1,200 @@
+// @flow
+/**
+ *
+ * Typeahead
+ *
+ */
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import classNames from 'classnames';
+
+import Input from '../../atoms/Input';
+import Label from '../../atoms/Label';
+import List from '../List';
+import ListItem from './ListItem';
+import styles from './Typeahead.style';
+import type { Props } from './types';
+import { ENTER_KEY, UP_ARROW, DOWN_ARROW, ESCAPE, TAB } from './constants';
+
+const Typeahead = ({
+  id,
+  className,
+  searchValue,
+  label,
+  isFieldLabel,
+  results,
+  onChangeHandler,
+  getResults,
+  renderCustomResultsView,
+  getSelectedResultValue,
+  customFilter,
+  ...otherProps
+}: Props): React.Element<any> => {
+  const [updatedResults, setFilteredResults] = useState([]);
+  const [activeResult, setActiveResult] = useState(0);
+
+  const triggerGetResults = val => {
+    return val && val.trim().length >= 3;
```

Comment:
This value 3 should be driven from Props, what if someone wants to set or not set this value.